src/Encoding/FilteredStream.php

```diff
@@ -57,10 +57,22 @@
      */
     public function __construct(StreamInterface $stream, $readFilterOptions = null, $writeFilterOptions = null)
     {
+        $triggerDeprecationNotice = null !== $writeFilterOptions;
+
+        if (null === $readFilterOptions && in_array($this->readFilter(), ['convert.base64-encode', 'convert.base64-decode', 'convert.quoted-printable-encode', 'zlib.deflate', 'zlib.inflate'/*, 'bzip2.compress', 'bzip2.decompress'*/], true)) {
+            $readFilterOptions = [];
+        }
+
+        if (null === $writeFilterOptions && in_array($this->writeFilter(), ['convert.base64-encode', 'convert.base64-decode', 'convert.quoted-printable-encode', 'zlib.deflate', 'zlib.inflate'/*, 'bzip2.compress', 'bzip2.decompress'*/], true)) {
+            $writeFilterOptions = [];
+
+            $triggerDeprecationNotice = false;
+        }
+
         $this->readFilterCallback = Filter\fun($this->readFilter(), $readFilterOptions);
         $this->writeFilterCallback = Filter\fun($this->writeFilter(), $writeFilterOptions);
 
-        if (null !== $writeFilterOptions) {
+        if ($triggerDeprecationNotice) {
```

Comment:
We should always trigger a deprecation notice if the `$writeFilterOptions` is used. The `$this->writeFilterCallback` is deprecated because it is never used.